src/huggingface_hub/cli/hf.py

```diff
@@ -12,51 +12,70 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from argparse import ArgumentParser
 
-from huggingface_hub.cli.auth import AuthCommands
-from huggingface_hub.cli.cache import CacheCommand
-from huggingface_hub.cli.download import DownloadCommand
-from huggingface_hub.cli.jobs import JobsCommands
-from huggingface_hub.cli.lfs import LfsCommands
-from huggingface_hub.cli.repo import RepoCommands
-from huggingface_hub.cli.repo_files import RepoFilesCommand
-from huggingface_hub.cli.system import EnvironmentCommand, VersionCommand
-from huggingface_hub.cli.upload import UploadCommand
-from huggingface_hub.cli.upload_large_folder import UploadLargeFolderCommand
+from huggingface_hub.cli._cli_utils import typer_factory
+from huggingface_hub.cli.auth import auth_cli
+from huggingface_hub.cli.cache import cache_cli
+from huggingface_hub.cli.download import download
+from huggingface_hub.cli.jobs import jobs_cli
+from huggingface_hub.cli.lfs import lfs_enable_largefiles, lfs_multipart_upload
+from huggingface_hub.cli.repo import repo_cli
+from huggingface_hub.cli.repo_files import repo_files_cli
+from huggingface_hub.cli.system import env, version
+
+# from huggingface_hub.cli.jobs import jobs_app
+from huggingface_hub.cli.upload import upload
+from huggingface_hub.cli.upload_large_folder import upload_large_folder
+from huggingface_hub.utils import logging
+
+
+app = typer_factory(help="Hugging Face Hub CLI")
+
+
+# top level single commands (defined in their respective files)
+app.command(
+    name="download",
+    help="Download files from the Hub.",
+)(download)
+app.command(
+    name="upload",
+    help="Upload a file or a folder to the Hub.",
+)(upload)
+app.command(
+    name="upload-large-folder",
+    help="Upload a large folder to the Hub. Recommended for resumable uploads.",
+)(upload_large_folder)
+app.command(
+    name="env",
+    help="Print information about the environment.",
+)(env)
+app.command(
+    name="version",
+    help="Print information about the hf version.",
+)(version)
+app.command(
+    name="lfs-enable-largefiles",
+    help="Configure your repository to enable upload of files > 5GB.",
+    hidden=True,
+)(lfs_enable_largefiles)
+app.command(
+    name="lfs-multipart-upload",
+    help="Upload large files to the Hub.",
+    hidden=True,
+)(lfs_multipart_upload)
```

Comment:
```suggestion
app.command(help="Download files from the Hub.")(download)
app.command(help="Upload a file or a folder to the Hub.")(upload)
app.command(help="Upload a large folder to the Hub. Recommended for resumable uploads.")(upload_large_folder)
app.command(help="Print information about the environment.")(env)
app.command(help="Print information about the hf version.")(version)
app.command(help="Configure your repository to enable upload of files > 5GB.", hidden=True)(lfs_enable_largefiles)
app.command(help="Upload large files to the Hub.", hidden=True)(lfs_multipart_upload)
```

Tested locally and it works to use implicit `name` based on variable name (which makes the whole definition much shorted)